### PR TITLE
move setting to config file, add btc feed, add binance and kraken

### PR DESCRIPTION
**Pull request by ductm54**
fetcher

### CODE REVIEW

File: feed-provider/common/orderbook.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			return orderbooks[p].Price > orderbooks[q].Price
+		}
+		return orderbooks[p].Price < orderbooks[q].Price
+	})
+	for _, o := range orderbooks {
+		size := o.Size
+		if remain < o.Price*size {
+			size = remain / o.Price
+		}
+		value := o.Price * size
```
Comment (secmask) [inline feed-provider/common/orderbook.go:54]: could you comment on calculate method here?

File: feed-provider/fetcher/binance/fetcher.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		return respBody, err
+	}
+	respBody, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return respBody, err
+	}
+	_ = resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return respBody, &common.RespError{
+			Msg:        "coinbase return not OK code",
```
Comment (secmask) [inline feed-provider/fetcher/binance/fetcher.go:68]: should be `binance` here

File: feed-provider/fetcher/binance/fetcher.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	req.Header.Add("Accept", "application/json")
+
+	q := req.URL.Query()
+	for k, v := range params {
+		q.Add(k, v)
+	}
+	req.URL.RawQuery = q.Encode()
+
+	f.sugar.Infof("request to coinbase: %s", req.URL)
```
Comment (secmask) [inline feed-provider/fetcher/binance/fetcher.go:56]: should be `binance`

File: feed-provider/fetcher/binance/fetcher.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		return respData, err
+	}
+	return respData, nil
+}
+
+// GetData get orderbook from coinbase and convert to feed data
+func (f *Fetcher) GetData() common.Feed {
+	resp, err := f.getData()
+	if err != nil {
+		f.sugar.Errorw("Get error while get coinbase feed", "error", err)
```
Comment (secmask) [inline feed-provider/fetcher/binance/fetcher.go:92]: should be `binance`

File: feed-provider/fetcher/binance/fetcher.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+		return common.Feed{
+			Error: err.Error(),
+		}
+	}
+	f.sugar.Debugw("Response from coinbase", "resp", resp)
```
Comment (secmask) [inline feed-provider/fetcher/binance/fetcher.go:97]: should be `binance`

File: feed-provider/fetcher/config.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+// NewConfigFromCli load config from file and create new instance
+func NewConfigFromCli(c *cli.Context) (*Configs, error) {
+	fileLocation := c.String(configLocationFlag)
+	jsonFile, err := os.Open(fileLocation)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+	byteValue, _ := ioutil.ReadAll(jsonFile)
```
Comment (secmask) [inline feed-provider/fetcher/config.go:49]: there's a shortcut function `ioutil.ReadFile(..)` so we dont have to do `os.Open` ourself.

File: feed-provider/fetcher/kraken/fetcher.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	}
+	req.Header.Add("Accept", "application/json")
+
+	q := req.URL.Query()
+	for k, v := range params {
+		q.Add(k, v)
+	}
+	req.URL.RawQuery = q.Encode()
+
+	f.sugar.Infof("request to coinbase: %s", req.URL)
```
Comment (secmask) [inline feed-provider/fetcher/kraken/fetcher.go:56]: copy/paste coinbase

File: feed-provider/fetcher/kraken/fetcher.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+		return respBody, err
+	}
+	respBody, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return respBody, err
+	}
+	_ = resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return respBody, &common.RespError{
+			Msg:        "coinbase return not OK code",
```
Comment (secmask) [inline feed-provider/fetcher/kraken/fetcher.go:68]: copy/paste coinbase

File: feed-provider/fetcher/kraken/fetcher.go [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+		return respData, err
+	}
+	return respData, nil
+}
+
+// GetData get orderbook from coinbase and convert to feed data
+func (f *Fetcher) GetData() common.Feed {
+	resp, err := f.getData()
+	if err != nil {
+		f.sugar.Errorw("Get error while get coinbase feed", "error", err)
```
Comment (secmask) [inline feed-provider/fetcher/kraken/fetcher.go:92]: copy/paste coinbase

File: feed-provider/fetcher/config.go (hunk 6 above)
Comment (ductm54) [reply feed-provider/fetcher/config.go]: fixed

File: feed-provider/fetcher/binance/fetcher.go (hunk 5 above)
Comment (ductm54) [reply feed-provider/fetcher/binance/fetcher.go]: fixed

File: feed-provider/fetcher/binance/fetcher.go (hunk 4 above)
Comment (ductm54) [reply feed-provider/fetcher/binance/fetcher.go]: fixed

File: feed-provider/fetcher/binance/fetcher.go (hunk 3 above)
Comment (ductm54) [reply feed-provider/fetcher/binance/fetcher.go]: fixed

File: feed-provider/fetcher/binance/fetcher.go (hunk 2 above)
Comment (ductm54) [reply feed-provider/fetcher/binance/fetcher.go]: fixed

File: feed-provider/common/orderbook.go (hunk 1 above)
Comment (ductm54) [reply feed-provider/common/orderbook.go:56]: add

File: feed-provider/fetcher/kraken/fetcher.go (hunk 7 above)
Comment (ductm54) [reply feed-provider/fetcher/kraken/fetcher.go]: fixed

File: feed-provider/fetcher/kraken/fetcher.go (hunk 8 above)
Comment (ductm54) [reply feed-provider/fetcher/kraken/fetcher.go]: fixed

File: feed-provider/fetcher/kraken/fetcher.go (hunk 9 above)
Comment (ductm54) [reply feed-provider/fetcher/kraken/fetcher.go]: fixed

**Pull request merged**